resources/migrations/001_update_migrations.yaml

```diff
@@ -5529,6 +5529,108 @@ databaseChangeLog:
             sql: INSERT INTO setting ("KEY", "VALUE") VALUES ('enable-public-sharing', 'false');
       rollback: # not needed
 
+  - changeSet:
+      id: v50.2024-02-16T16:00:00
+      author: piranha
+      comment: 'Effective caching #36847'
+      changes:
+        - createTable:
+            tableName: cache_config
+            remarks: Cache Configuration
+            columns:
+              - column:
+                  name: id
+                  remarks: Unique ID
+                  type: int
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: model
+                  remarks: Name of an entity model
+                  type: varchar(32)
+                  constraints:
+                    nullable: true
+              - column:
+                  name: model_id
+                  remarks: ID of the said entity
+                  type: int
+                  constraints:
+                    nullable: true
+              - column:
```

Comment:
what is collection for in this context? is it used for collection level cache config? if so I don't see it's specced in the product doc

Reply:
It's to realize nesting in the UI. So when a request comes in asking for everything inside the collection I don't end up joining 5 different tables with a big case to understand what needs to be selected from a db.

OTOH I need some bookkeeping for when things are moved between collections then?... :\

Reply:
> OTOH I need some bookkeeping for when things are moved between collections then?... :\

yes :)

Reply:
also, I think you need to think about the fact that collection are also nested.

so with this, at most, you can only get items one level deep.

An idea is to bake this into `GET /collection/:id/items` by adding an `include_cache_config` flag, then we can hydrate it for all items.

anw, I think we need to align with FE on how they want cache config to be returned here.

Reply:
> so with this, at most, you can only get items one level deep.

how so? :) if I get a collection_id, they can be nested in any collection they want how deep they want.

> An idea is to bake this into GET /collection/:id/items by adding an include_cache_config flag, then we can hydrate it for all items.

I've been pondering about that, but it felt like adding too much on top of it's functionality?.. IDK maybe that's better way to go.